torchtitan/config_manager.py

```diff
@@ -532,10 +532,22 @@ def _args_to_two_level_dict(self, args: argparse.Namespace) -> defaultdict:
             args_dict[first_level_key][second_level_key] = v
         return args_dict
 
-    def _validate_config(self) -> bool:
+    def _validate_config(self) -> None:
         # TODO: Add more mandatory validations
-        assert self.model.name and self.model.flavor and self.model.tokenizer_path
-        return True
+        assert self.model.name
+        assert self.model.flavor
+        assert self.model.tokenizer_path
+
+        ac_config = self.activation_checkpoint
+        ac_config = self.activation_checkpoint
+        if ac_config.mode not in ("full", "selective", "none"):
+            raise ValueError(f"Invalid AC mode: {ac_config.mode}")
```

Comment:
I think putting config checks here makes it easier to understand code. I have two concerns though:
1. We have many similar configs to be checked, e.g. `training.norm_type`. Shall we all follow this style -- having a centralized place for checks.
2. Config check is put far from the config usage, which could cause confusion/inconsistency when making changes to the configs themselves.

Given the two reasons above, I personally prefer putting the checks back to where they are used. What do you think?